### ISSUE TITLE
What loss function is being used?

### ISSUE THREAD

**Issue by kitchWWW** [archinetai/audio-diffusion-pytorch#53]
Hi! Working through getting a working version of this set up locally and running it on some of my own data. The paradigm I see with a lot of models is something along the lines of:

```
for input, target in dataset:
    optimizer.zero_grad()
    output = model(input)
    loss = loss_fn(output, target)
    loss.backward()
    optimizer.step()
# taken from https://pytorch.org/docs/stable/optim.html
```

Where an output is returned from the model, and then the loss computed with a loss function, before we call a `optimizer.step()` function. However, in all of the examples you provide, the sample is something like:
```
loss = model(input)
loss.backward()
```
What loss function is being used here? Is it being inherited from `a-unet` or one of the parent classes somewhere along the line? TIA.

**Comment by flavioschneider** [comment]
The loss function used is a MSE on the predicted noise rate of change, see here:
https://github.com/archinetai/audio-diffusion-pytorch/blob/a06b92a299f8a266f2144c36f7f05502ffed845b/audio_diffusion_pytorch/diffusion.py#L94
